File: tests/adapters/java/goldens/imports/strip_local.java
/**
 * Java module for testing import optimization.
 */

package com.example.imports;

// Standard library imports (external)
import java.util.*;
import java.util.concurrent.*;
import java.util.stream.*;
import java.time.*;
import java.time.format.*;
import java.nio.file.*;
import java.io.*;

// More standard library
import java.math.BigDecimal;
import java.math.BigInteger;
import java.net.URI;
import java.net.URL;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

// Java EE / Jakarta EE imports (external)
import javax.validation.constraints.*;
import javax.validation.Valid;
import javax.persistence.*;
import javax.servlet.http.*;
import javax.annotation.*;

// Third-party library imports (external)
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.transaction.annotation.Transactional;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnore;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.apache.commons.lang3.StringUtils;
import org.apache.commons.collections4.CollectionUtils;
import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableMap;

// Local/relative imports (should be considered local)
// … 9 imports omitted

// Imports from different package levels
// … 3 imports omitted

// Static imports
import static java.util.Collections.*;
import static java.util.stream.Collectors.*;
import static org.springframework.http.HttpStatus.*;
// … import omitted

// Long import lists from single package (candidates for summarization)
import org.springframework.http.ResponseEntity;
import org.springframework.http.HttpStatus;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.http.HttpMethod;
import org.springframework.http.RequestEntity;
import org.springframework.http.converter.HttpMessageConverter;
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
import org.springframework.http.converter.xml.MappingJackson2XmlHttpMessageConverter;
import org.springframework.http.server.ServerHttpRequest;
import org.springframework.http.server.ServerHttpResponse;
import org.springframework.http.server.ServletServerHttpRequest;

// … 23 imports omitted

@Service
public class ImportTestService {
    private static final Logger logger = LoggerFactory.getLogger(ImportTestService.class);

    @Autowired
    private UserService userService;

    @Autowired
    private DatabaseConnection dbConnection;

    @Autowired
    private ObjectMapper objectMapper;

    public ApiResponse<List<Map<String, Object>>> processData(List<Object> data) throws JsonProcessingException {
        // Using external libraries
        List<Map<String, Object>> processed = data.stream()
            .map(item -> {
                Map<String, Object> result = new HashMap<>();
                result.put("id", UUID.randomUUID().toString());
                result.put("timestamp", Instant.now().toString());
                result.put("item", item);
                return result;
            })
            .collect(toList());

        // Using local utilities
        List<Map<String, Object>> validated = processed.stream()
            .filter(item -> {
                String email = (String) item.get("email");
                return email != null && EmailValidator.validate(email);
            })
            .collect(toList());

        // Using Java NIO
        Path filePath = Paths.get("output.json");
        String json = objectMapper.writeValueAsString(validated);
        Files.writeString(filePath, json);

        return new ApiResponse<>(
            true,
            validated,
            DateFormatter.format(LocalDateTime.now())
        );
    }

    public String makeHttpRequest(String url) throws IOException, InterruptedException {
        try {
            // Using Java 11+ HTTP Client
            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .timeout(Duration.ofSeconds(5))
                .header("User-Agent", "ImportTestService/1.0")
                .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            return response.body();
        } catch (Exception e) {
            logger.error("HTTP request failed", e);
            throw new NetworkError("Request failed");
        }
    }

    public String serializeData(Object data) throws JsonProcessingException {
        // Using Jackson
        objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
        return objectMapper.writeValueAsString(data);
    }
}

// Annotated classes using validation imports
@Entity
@Table(name = "users")
public class ValidatedUser {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    private Long id;

    @NotBlank
    @Size(min = 2, max = 100)
    @Column(nullable = false)
    private String name;

    @Email
    @NotEmpty
    @Column(nullable = false, unique = true)
    private String email;

    @Positive
    private Integer age;

    @Pattern(regexp = "^\\+?[1-9]\\d{1,14}$")
    private String phone;

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public Integer getAge() { return age; }
    public void setAge(Integer age) { this.age = age; }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }
}

// REST Controller using Spring annotations
@RestController
@RequestMapping("/api/users")
public class UserController {
    @Autowired
    private ImportTestService importTestService;

    @GetMapping("/{id}")
    public ResponseEntity<ValidatedUser> getUser(@PathVariable Long id) {
        // Implementation
        return ResponseEntity.ok().build();
    }

    @PostMapping
    public ResponseEntity<ValidatedUser> createUser(@Valid @RequestBody ValidatedUser user) {
        // Implementation
        return ResponseEntity.status(CREATED).build();
    }

    @PutMapping("/{id}")
    public ResponseEntity<ValidatedUser> updateUser(
        @PathVariable Long id,
        @Valid @RequestBody ValidatedUser user
    ) {
        // Implementation
        return ResponseEntity.ok().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteUser(@PathVariable Long id) {
        // Implementation
        return ResponseEntity.noContent().build();
    }
}

File: tests/adapters/kotlin/goldens/imports/strip_local.kt
/**
 * Kotlin module for testing import optimization.
 */

package com.example.imports

// Standard library imports (external)
import kotlin.math.*
import kotlin.collections.*
import kotlin.text.Regex
import kotlin.io.path.Path
import kotlin.random.Random
import java.time.Instant
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import java.util.UUID
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.atomic.AtomicInteger

// Third-party library imports (external)
import kotlinx.coroutines.*
import kotlinx.coroutines.flow.*
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import io.ktor.client.*
import io.ktor.client.request.*
import io.ktor.client.statement.*
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import org.slf4j.Logger
import org.slf4j.LoggerFactory

// … 17 imports omitted (21 lines)

// Long import lists (candidates for summarization)
import io.ktor.server.application.Application
import io.ktor.server.application.call
import io.ktor.server.application.install
import io.ktor.server.response.respond
import io.ktor.server.response.respondText
import io.ktor.server.routing.Route
import io.ktor.server.routing.get
import io.ktor.server.routing.post
import io.ktor.server.routing.put
import io.ktor.server.routing.delete
import io.ktor.server.routing.routing
import io.ktor.http.HttpStatusCode
import io.ktor.http.Headers
import io.ktor.http.ContentType
import io.ktor.http.Parameters

import org.koin.core.component.KoinComponent
import org.koin.core.component.inject
import org.koin.core.context.startKoin
import org.koin.core.context.stopKoin
import org.koin.core.module.Module
import org.koin.core.module.dsl.singleOf
import org.koin.core.module.dsl.factoryOf
import org.koin.core.qualifier.named
import org.koin.dsl.module
import org.koin.dsl.bind
import org.koin.dsl.binds

import javax.validation.constraints.NotNull
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Email
import javax.validation.constraints.Size
import javax.validation.constraints.Min
import javax.validation.constraints.Max
import javax.validation.constraints.Pattern
import javax.validation.constraints.Positive
import javax.validation.constraints.PositiveOrZero
import javax.validation.constraints.Negative
import javax.validation.constraints.NegativeOrZero
import javax.validation.Valid

// … 23 imports omitted (24 lines)

class ImportTestService(
    private val userService: UserService,
    private val dbConnection: DatabaseConnection,
    private val logger: AppLogger
) : KoinComponent {
    
    suspend fun processData(data: List<Any>): ApiResponse<List<Any>> {
        // Using external libraries
        val processed = data.map { item ->
            mapOf(
                "id" to UUID.randomUUID(),
                "timestamp" to Instant.now().toString(),
                "item" to item
            )
        }
        
        // Using local utilities
        val validated = processed.mapNotNull { item ->
            val email = item["email"] as? String
            if (email != null && validateEmail(email)) item else null
        }
        
        // Using Kotlin standard library
        val filePath = Path("/output.json")
        
        return ApiResponse(
            success = true,
            data = validated,
            timestamp = formatDate(LocalDateTime.now())
        )
    }
    
    suspend fun makeHttpRequest(url: String): String? {
        return try {
            // Using Ktor client
            val client = HttpClient()
            val response: HttpResponse = client.get(url) {
                timeout {
                    requestTimeoutMillis = 5000
                }
                headers {
                    append("User-Agent", "ImportTestService/1.0")
                }
            }
            
            response.bodyAsText()
        } catch (e: Exception) {
            logger.error("HTTP request failed", e)
            throw NetworkError("Request failed")
        }
    }
    
    fun serializeData(data: Any): String {
        // Using Gson
        val gson = GsonBuilder()
            .setPrettyPrinting()
            .create()
        
        return gson.toJson(data)
    }
}

// Annotated classes using validation imports
@Serializable
data class ValidatedUser(
    @NotNull
    @NotEmpty
    val id: String,
    
    @NotBlank
    @Size(min = 2, max = 100)
    val name: String,
    
    @Email
    @NotEmpty
    val email: String,
    
    @Positive
    val age: Int?,
    
    @Pattern(regexp = "^\\+?[1-9]\\d{1,14}$")
    val phone: String?
)

// Function using multiple imports
suspend fun processWithCoroutines(items: List<String>): Flow<String> = flow {
    items.forEach { item ->
        delay(100)
        emit(item.uppercase())
    }
}.flowOn(Dispatchers.Default)

// Koin module definition
val appModule = module {
    single { DatabaseConnection(get()) }
    singleOf(::UserService) bind AppLogger::class
    factory { ImportTestService(get(), get(), get()) }
}

// Default export
fun main() {
    startKoin {
        modules(appModule)
    }
    
    val service: ImportTestService by inject()
    println("Service initialized")
}
